kafka-connect/kafka-connect/src/test/java/org/apache/iceberg/connect/data/IcebergWriterFactoryTest.java

```diff
@@ -47,7 +54,7 @@ public class IcebergWriterFactoryTest {
   @ValueSource(booleans = {true, false})
   @SuppressWarnings("unchecked")
   public void testAutoCreateTable(boolean partitioned) {
-    Catalog catalog = mock(Catalog.class);
+    Catalog catalog = mock(InMemoryCatalog.class);
```

Comment:
Need to change it to some catalog that supports namespace creation. As the code now calls `createNamespaceIfNotExist`

Reply:
I'd rather not rely on InMemoryCatalog API here.  You could do something like this instead, `Catalog catalog = mock(Catalog.class, withSettings().extraInterfaces(SupportsNamespaces.class));`

Reply:
done